File: 2021/day01/day1.pl
open STDIN, "day1.txt";


($l<$_&&$a++),$l=$_ while<>;
print "part1: $a";

__END__

open STDIN, "day1.txt";

my $acc = 0;
my $last;

while (<>) {
 $acc++ if $last < $_;
 $last = $_;
}

print "part1: $acc";

File: 2021/day01/day1.rb
lines = open('day1.txt').map(&:to_i)

puts lines
  .each_cons(2)
  .count { _1 < _2 }

puts lines
  .each_cons(3)
  .map(&:sum)
  .each_cons(2)
  .count { _1 < _2 }
